src/protocol_handle_stream.rs

```diff
@@ -285,36 +285,31 @@ where
             return Poll::Ready(None);
         }
 
-        loop {
-            match Pin::new(&mut self.receiver).as_mut().poll_next(cx) {
-                Poll::Ready(Some(event)) => self.handle_event(event),
-                Poll::Ready(None) => {
-                    debug!(
-                        "ServiceProtocolStream({:?}) finished",
-                        self.handle_context.proto_id
-                    );
-                    self.current_task.idle();
-                    return Poll::Ready(None);
-                }
-                Poll::Pending => {
-                    break;
-                }
+        let mut is_pending = false;
```

Comment:
This poll contains 3 sub polls. From the std lib document, we know that the poll should only return Pending when at least one of the sub poll returns Pending. But what's happen when one of the sub poll returns Pending and the parent poll returns Ready?

Reply:
Whenever the stream returns ready, it will immediately enter the next poll